databases/postgresql.md

```diff
@@ -0,0 +1,259 @@
+# PostgreSQL
+
+[PostgreSQL](http://www.postgresql.org/) is an object-relational database management system (ORDBMS).
+
+The connection between PostgreSQL and Ruby happens via the [`pg`](https://rubygems.org/gems/pg) gem.
+
+## Installation
+
+[Install PostgreSQL](http://www.postgresql.org/download/)
+
+```
+gem install pg
+```
+
+**ActiveRecord or DataMapper**. You need only one, but we'll show both.
+
+
+```
+gem install activerecord
+gem install sinatra-activerecord
+gem install rake # to apply migrations
+# or
+gem install dm-core
+gem install dm-postgres-adapter
+```
+
+
+## ActiveRecord usage
+
+**Connect to the database**
+
+Either by making a `config/database.yml` file that gets automatically loaded
+
+```
+development:
+  adapter: postgresql
+  encoding: unicode
+  database: database_name
+  pool: 2
+  username: your_username
+  password: your_password
+
+production:
+  adapter: postgresql
+  encoding: unicode
+  database: database_name
+  pool: 2
+  username: your_username
+  password: your_password
+```
+
+or by specifying in your sinatra app
+
+```ruby
+configure :development do
+  set :database, {adapter: 'postgres',  encoding: 'unicode', database: 'your_database_name', pool: 2, username: 'your_username', password: 'your_password'}
+end
+
+configure :production do
+  set :database, {adapter: 'postgres',  encoding: 'unicode', database: 'your_database_name', pool: 2, username: 'your_username', password: 'your_password'}
+end
+```
+
+**Create models**
+```ruby
+class Article < ActiveRecord::Base
+end
+```
+
+**Create migrations**
+
+First, create a `Rakefile` or update your `Rakefile` to require `activerecord/rake`
+
+```
+require 'sinatra/activerecord/rake'
+...
```

Comment:
The require line is needed to expose the `sinatra-activerecord` gem rake functionality.
```
rake db:create
rake db:drop
rake db:migrate
..... and etc
```